core/common/test/LocalTimeTest.kt

```diff
@@ -85,6 +83,58 @@ class LocalTimeTest {
         assertFailsWith<IllegalArgumentException> { LocalTime(0, 0, 0, 1_000_000_000) }
     }
 
+    @Test
+    fun fromNanosecondOfDay() {
+        val data = mapOf(
+            0L to LocalTime(0, 0),
+            5000000001L to LocalTime(0, 0, 5, 1),
+            44105000000100L to LocalTime(12, 15, 5, 100),
+            86399000000999L to LocalTime(23, 59, 59, 999),
```

Comment:
All of these have the millisecond and microsecond triads zeroed. I'd add a test with `123456789` nanoseconds or something like that.